src/peergos/server/space/JdbcQuotas.java

```diff
@@ -0,0 +1,149 @@
+        }
+    }
+
+    public void setQuota(String username, long quota) {
+        try (Connection conn = getConnection();
+             PreparedStatement createuser = conn.prepareStatement(commands.insertOrIgnoreCommand("INSERT ", "INTO freequotas (name, quota) VALUES(?, ?)"));
+             PreparedStatement update = conn.prepareStatement(SET_QUOTA)) {
+            createuser.setString(1, username);
+            createuser.setLong(2, 0);
+            createuser.executeUpdate();
+
+            update.setLong(1, quota);
+            update.setString(2, username);
+            update.executeUpdate();
+        } catch (SQLException sqe) {
+            LOG.log(Level.WARNING, sqe.getMessage(), sqe);
+            throw new IllegalStateException(sqe);
+        }
+    }
+
+    public long getQuota(String username) {
+        try (Connection conn = getConnection();
+             PreparedStatement select = conn.prepareStatement(GET_QUOTA)) {
+            select.setString(1, username);
+            ResultSet rs = select.executeQuery();
+            if (rs.next())
+                return rs.getLong(QUOTA_SIZE);
+
+            throw new IllegalStateException("No quota listed for user!");
+        } catch (SQLException sqe) {
+            LOG.log(Level.WARNING, sqe.getMessage(), sqe);
+            throw new IllegalStateException(sqe);
+        }
+    }
+
+    public void removeUser(String username) {
+        try (Connection conn = getConnection();
+             PreparedStatement delete = conn.prepareStatement(REMOVE_USER)) {
+            delete.setString(1, username);
+            delete.executeUpdate();
+        } catch (SQLException sqe) {
+            LOG.log(Level.WARNING, sqe.getMessage(), sqe);
+            throw new IllegalStateException(sqe);
+        }
+    }
+
+    public Map<String, Long> getQuotas() {
+        try (Connection conn = getConnection();
+             PreparedStatement select = conn.prepareStatement(GET_ALL_QUOTAS)) {
+            ResultSet rs = select.executeQuery();
+            Map<String, Long> res = new HashMap<>();
+            while (rs.next()) {
+                String username = rs.getString(QUOTA_USER_NAME);
+                long quota = rs.getLong(QUOTA_SIZE);
+                res.put(username, quota);
+            }
+            return res;
+        } catch (SQLException sqe) {
+            LOG.log(Level.WARNING, sqe.getMessage(), sqe);
+            throw new IllegalStateException(sqe);
+        }
+    }
+
+    public boolean hasUser(String username) {
+        try (Connection conn = getConnection();
+             PreparedStatement select = conn.prepareStatement(GET_QUOTA)) {
+            select.setString(1, username);
+            ResultSet rs = select.executeQuery();
+            return rs.next();
+        } catch (SQLException sqe) {
+            LOG.log(Level.WARNING, sqe.getMessage(), sqe);
+            throw new IllegalStateException(sqe);
+        }
+    }
+
+    public int numberOfUsers() {
```

Comment:
select count() is the general way to do that in SQL

Reply:
done